definitions/checks/foreman/check_checkpoint_segments.rb

```diff
@@ -0,0 +1,50 @@
+module Checks
+  module Foreman
+    class CheckpointSegments < ForemanMaintain::Check
+      metadata do
+        label :check_checkpoint_segments
+        description 'Check if checkpoint_segments parameter persent in custom hiera'
+        tags :pre_upgrade
+      end
+
+      HIERA_FILE = '/etc/foreman-installer/custom-hiera.yaml'.freeze
+
+      def run
+        files = []
+        files << feature(:foreman_database).config_files[0] if check_postgres_config
+        files << HIERA_FILE if check_custom_hiera
+        unless files.empty?
+          message = ERB.new(<<-BLOCK.strip_heredoc).result(binding)
+          ERROR:
+            checkpoint_segments tuning found.
+            This tuning option is no longer valid in PostgreSQL 9.5+
+            Please remove this from the following locations and then re-run the foreman-maintain upgrade check:
+            <% files.each do |file| %>
+              - <%= file %>
+            <% end %>
+          BLOCK
+          fail! message
+        end
+      end
+
+      def check_custom_hiera
```

Comment:
After defining a method inside `installer` feature, you can use a variable instead of constant.
`hiera_file = feature(:installer) ? feature(:installer).custom_hiera_file : nil`